Sources/SwiftJWT/JWTEncoder.swift

```diff
@@ -150,10 +133,21 @@ fileprivate class _JWTEncoder: Encoder {
         mutating func encode<T : Encodable>(_ value: T, forKey key: Key) throws {
```

Comment:
Similar to `decode`, I think this (or at least the `_JWTEncoder` type warrants comments explaining how it works (the execution flow) for the benefit of future maintainers.
For example, we are stashing the signer on the encoder so that it can set the `alg` header appropriately when the headers are encoded - it would be worth a comment as such, and also to explain why we would want to do this.